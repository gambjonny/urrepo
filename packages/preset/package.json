{
  "name": "@urrepo/preset",
  "type": "module",
  "version": "0.0.1",
  "description": "core preset",
  "main": "src/index.ts",
  "sideEffects": false,
  "keywords": [
    "panda-css",
    "preset",
    "theme",
    "styling"
  ],
  "scripts": {
    "build": "wireit",
    "test": "wireit",
    "test:watch": "vitest",
    "coverage": "wireit",
    "lint:check": "wireit",
    "lint:fix": "wireit",
    "format:check": "wireit",
    "format:fix": "wireit"
  },
  "engines": {
    "node": "22.12.0",
    "pnpm": "10.0.0"
  },
  "volta": {
    "extends": "../../package.json"
  },
  "devDependencies": {
    "@types/node": "catalog:",
    "@urrepo/eslint-config": "workspace:*",
    "@vitest/coverage-v8": "catalog:",
    "@vitest/ui": "catalog:",
    "jiti": "catalog:",
    "typescript": "catalog:",
    "vitest": "catalog:"
  },
  "peerDependencies": {
    "@pandacss/dev": "catalog:"
  },
  "wireit": {
    "build": {
      "command": "tsc",
      "dependencies": [],
      "files": [
        "**/*.{ts,tsx,js,json}",
        "*.config.*"
      ],
      "output": [
        "dist/**"
      ]
    },
    "test": {
      "command": "vitest run",
      "files": [
        "src/**/*.ts",
        "tests/**/*.ts",
        "vitest.config.ts"
      ]
    },
    "coverage": {
      "command": "vitest run --coverage",
      "dependencies": [
        "test"
      ]
    },
    "lint:check": {
      "command": "eslint .",
      "files": [
        "src/**/*.ts",
        "tests/**/*.ts",
        "eslint.config.ts"
      ]
    },
    "lint:fix": {
      "command": "eslint . --fix",
      "files": [
        "src/**/*.ts",
        "tests/**/*.ts",
        "eslint.config.ts"
      ]
    },
    "format:check": {
      "command": "prettier --ignore-path ../../.prettierignore --check .",
      "files": [
        "src/**/*.ts",
        "tests**/*.ts"
      ]
    },
    "format:fix": {
      "command": "prettier --ignore-path ../../.prettierignore --write .",
      "files": [
        "src/**/*.ts",
        "tests/**/*.ts"
      ]
    }
  }
}
