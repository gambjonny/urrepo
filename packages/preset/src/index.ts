import { definePreset } from '@pandacss/dev'

const preset = definePreset({
  name: 'core',
  theme: {
    extend: {
      tokens: {
        colors: {
          cobalt: {
            50: {
              value: '#ebedf9',
            },
            100: {
              value: '#ebedf9',
            },
            200: {
              value: '#ebedf9',
            },
            300: {
              value: '#ebedf9',
            },
            400: {
              value: '#ebedf9',
            },
          },
        },
      },
    },
  },
})

export default preset
