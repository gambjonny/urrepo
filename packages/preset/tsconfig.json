{
  "extends": "../../tsconfig.base.json",
  "include": ["src/**/*", "tests/**/*", "vitest.config.ts"],
  "compilerOptions": {
    "noEmit": true,
    "baseUrl": "."
  }
}
