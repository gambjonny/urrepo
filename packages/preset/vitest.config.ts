import { defineProject } from "vitest/config";

export default defineProject({
  test: {
    globals: true,
    environment: "node",
    include: ["tests/**/*.ts"],
  },
});
