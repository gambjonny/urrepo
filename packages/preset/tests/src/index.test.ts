import preset from "../../src/index";

describe.concurrent("PandaCSS Preset", () => {
  it("matches the snapshot", ({ expect }) => {
    expect(preset).toMatchSnapshot();
  });
});
