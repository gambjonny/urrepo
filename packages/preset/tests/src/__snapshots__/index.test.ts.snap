// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PandaCSS Preset > matches the snapshot 1`] = `
{
  "name": "core",
  "theme": {
    "extend": {
      "tokens": {
        "colors": {
          "cobalt": {
            "100": {
              "value": "#D3D7EE",
            },
            "200": {
              "value": "#959CDE",
            },
            "300": {
              "value": "#5460C8",
            },
            "400": {
              "value": "#1020B9",
            },
            "50": {
              "value": "#EBEDF9",
            },
          },
        },
      },
    },
  },
}
`;
